client/index.ts

```diff
@@ -0,0 +1,452 @@
+enum Role { SUPERADMIN, ADMIN, SUBSCRIBER };
+
+type NewUser={
+    firstName: string;
+    middleName: string;
+    lastName: string;
+    email: string;
+    phoneNumber: string;
+    role: Role;
+    address: string;
+
+}
+type User ={
+    id: number;
+    firstName: string;
+    middleName: string;
+    lastName: string;
+    email: string;
+    phoneNumber: string;
+    role: Role;
+    address: string;
+}
+let crudObject: Crud<User>;  //array of objects to be displayed on frontend
+
+//DECORATOR FACTORY
+function FormatDate() {
+    return function (target: any, name: string, descriptor: PropertyDescriptor) {
+        const dateTime = document.getElementById("dateTime")! as HTMLInputElement;
+        dateTime.innerHTML = new Date().toLocaleString();
+        setInterval(function () {
+            dateTime.innerHTML = new Date().toLocaleString();
+        }, 1000);
+    }
+}
+
+//MODEL CLASS FOR USER ENTRY
+class Crud<T> {
+
+    items: Array<T>; // contains objects which will be displayed on front end
+
+    constructor() {
+        this.items = [];
+    }
+    @FormatDate()
+    add(object: T): void {
+        this.items.push(object)
+    }
+
+    update(id: number, index: number, updatedObject: T) {
+
+        editUser(id, updatedObject).then((response) => {
+
+            this.items[index] = response.updatedRecord;
+            alert(response.message);
+            showTable()
+        }).catch(() => {
+            console.log('update');
+            alert("Unexpected Error Occured !")
+        })
+    }
+    delete(id: number, index: number, object: T): void {
+
+        //make api call to delete data 
+        deleteUser(id).then((response) => {
+            //delete data on front end
+
+            this.items.splice(index, 1);
```

Comment:
rather than splicing. refresh the data and show new data.

Reply:
Maam I can't do this.items = data returned by get request, as items is a generic Array ----- items: Array<T>;